IPython/core/tests/test_completerlib.py

```diff
@@ -77,21 +77,22 @@ def test_completion_more_args(self):
 
     def test_completion_in_dir(self):
         # Github issue #3459
-        event = MockEvent(u'%run a.py {}'.format(join(self.BASETESTDIR, 'a')))
+        event = MockEvent(u"%run a.py {}".format(self.BASETESTDIR.joinpath("a")))
         print(repr(event.line))
         match = set(magic_run_completer(None, event))
         # We specifically use replace here rather than normpath, because
         # at one point there were duplicates 'adir' and 'adir/', and normpath
         # would hide the failure for that.
-        self.assertEqual(match, {join(self.BASETESTDIR, f).replace('\\','/')
-                            for f in (u'a.py', u'aao.py', u'aao.txt', u'adir/')})
+        self.assertEqual(match, {str(self.BASETESTDIR.joinpath(f)) if Path(f).suffix
+                                 else str(self.BASETESTDIR.joinpath(f)) + os.sep
+                            for f in (u"a.py", u"aao.py", u"aao.txt", u"adir/")})
```

Comment:
Not sure it's the best way to do this, it works but seems more complicated than before... :thinking:

Reply:
Indeed it doesn't work on Windows... :sweat: